gobblin-runtime/src/main/java/gobblin/runtime/mapreduce/MRJobLauncher.java

```diff
@@ -594,7 +594,9 @@ protected void cleanup(Context context) throws IOException, InterruptedException
         // Ignored
       } finally {
         if (this.jobMetrics.isPresent()) {
+          this.jobMetrics.get().triggerMetricReporting();
           this.jobMetrics.get().stopMetricReporting();
+          JobMetrics.remove(this.jobMetrics.get().getName());
```

Comment:
Also suggesting put `remove` into a `finally` clause following a `catch` around the above two statements.